sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -507,6 +509,136 @@ case class GpuArrayMax(child: Expression) extends GpuUnaryExpression with Implic
     }
 }
 
+case class GpuArraysZip(children: Seq[Expression]) extends GpuExpression with ExpectsInputTypes {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq.fill(children.length)(ArrayType)
+
+  @transient override lazy val dataType: DataType = {
+    val fields = children.zip(arrayElementTypes).zipWithIndex.map {
+      case ((expr: NamedExpression, elementType), _) =>
+        StructField(expr.name, elementType, nullable = true)
+      case ((_, elementType), idx) =>
+        StructField(idx.toString, elementType, nullable = true)
+    }
+    ArrayType(StructType(fields), containsNull = false)
+  }
+
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  @transient private lazy val arrayElementTypes =
+    children.map(_.dataType.asInstanceOf[ArrayType].elementType)
+
+  override def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression =
+    legacyWithNewChildren(newChildren)
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    val numRows = batch.numRows()
+    val numCols = children.size
+    // A container collects all column vectors in use, to ease the release on exceptions.
+    val workBuffers = mutable.ArrayBuffer[ColumnVector]()
+
+    if (numCols == 0) {
+      return GpuScalar(new GenericArrayData(Array.empty[Any]), dataType)
+    }
+
+    // Prepare input columns
+    closeOnExcept(workBuffers) { _ =>
+      children.foreach { expr =>
+        workBuffers += GpuExpressionsUtils.columnarEvalToColumn(expr, batch).getBase
+      }
+    }
+    val inputs = workBuffers.toArray
```

Comment:
nit: This is very, very minor but technically this is what safeMap was written for.
```
val inputs = children.safeMap{ expr =>
  GpuExpressionsUtils.columnarEvalToColumn(expr, batch).getBase
}.toArray
```